backend/api/Controllers/RobotController.cs

```diff
@@ -214,26 +214,43 @@ public async Task<ActionResult<Robot>> PostRobot([FromBody] Robot robot)
             }
 
             _logger.LogError(e, "Error getting mission from Echo");
-            return new StatusCodeResult(StatusCodes.Status502BadGateway);
+            return StatusCode(StatusCodes.Status502BadGateway, $"{e.Message}");
         }
         catch (JsonException e)
         {
             _logger.LogError(e, "Error deserializing mission from Echo");
-            return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+            return StatusCode(StatusCodes.Status500InternalServerError, $"{e.Message}");
         }
 
+        Report report;
         try
         {
-            var report = await _isarService.StartMission(robot, echoMissionId, new IsarMissionDefinition(echoMission));
-            robot.Status = RobotStatus.Busy;
+            report = await _isarService.StartMission(robot, echoMissionId, new IsarMissionDefinition(echoMission));
+        }
+        catch (HttpRequestException e)
+        {
+            _logger.LogError(e, "Error connecting to ISAR while starting mission");
+            robot.Enabled = false;
+            robot.Status = RobotStatus.Offline;
             await _robotService.Update(robot);
-            return Ok(report);
+
+            return StatusCode(StatusCodes.Status502BadGateway, $"{e.Message}");
```

Comment:
Maybe here we could return our own message since we know what the issue is (ISAR is offline)
Same for other endpoints

Reply:
Good idea!